internal/k8s/pod/pod.go

```diff
@@ -237,10 +219,7 @@ func extractContainerID(str string) string {
 
 func (pi *podInformer) Run(ctx context.Context) error {
 	pi.logger.Info("Starting pod informer")
-	mgrCtx, cancel := context.WithCancel(context.Background())
-	pi.mgrCtx = mgrCtx
-	pi.mgrCancelFunc = cancel
-	return pi.manager.Start(mgrCtx)
```

Comment:
there are two ways to gracefully cleanup resources on api server. mgr.Stop() or cancel the context used in mgr.Start